mypy/config_parser.py

```diff
@@ -184,9 +242,84 @@ def parse_config_file(options: Options, set_strict_flags: Callable[[], None],
                     options.per_module_options[glob] = updates
 
 
+def destructure_overrides(toml_data: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+    """Take the new [[tool.mypy.overrides]] section array in the pyproject.toml file,
+    and convert it back to a flatter structure that the existing config_parser can handle.
+
+    E.g. the following pyproject.toml file:
+
+        [[tool.mypy.overrides]]
+        module = [
+            "a.b",
+            "b.*"
+        ]
+        disallow_untyped_defs = true
+
+        [[tool.mypy.overrides]]
+        module = 'c'
+        disallow_untyped_defs = false
+
+    Would map to the following config dict that it would have gotten from parsing an equivalent
+    ini file:
+
+        {
+            "mypy-a.b": {
+                disallow_untyped_defs = true,
+            },
+            "mypy-b.*": {
+                disallow_untyped_defs = true,
+            },
+            "mypy-c": {
+                disallow_untyped_defs: false,
+            },
+        }
+    """
+    if 'overrides' not in toml_data['mypy']:
+        return toml_data
+
+    if not isinstance(toml_data['mypy']['overrides'], list):
+        raise ValueError("tool.mypy.overrides sections must be an array. Please make sure you are "
+                         "using double brackets like so: [[tool.mypy.overrides]]")
```

Comment:
Raise a more specific error (e.g. define a new exception class), and make sure the errors are reported cleanly and don't generate a traceback. It would be good to have tests for the error cases.